File: algorithm/programmers/level_1/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        int com[] = new int[3];
        int cnt = 0;
        int len = array.length;
        for(int i=0; i<commands.length; i++){
            for(int j=0; j<commands[i].length; j++){
              com[j] = commands[i][j];
            } //command2
            int from = com[0];
            int to = com[1];
            int get = com[2];
            int temp_len = (to-from)+1;
            int temp[] = new int[temp_len];
            for(int k=1; k<=len; k++){
                int idx = 0;
                if(k==from){
                    while(temp_len!=0){
                        temp[idx++] = array[k-1];
                        from++;
                        k++;
                        cnt ++;
                        temp_len--;
                } //while 
                } //if
                else{continue;}
            }
            Arrays.sort(temp);
            for(int a=0; a<cnt; a++){
                if(a==get-1){
                     answer[i] = temp[a];
                }
                System.out.print("temp is :" + temp[a]);
            }
            System.out.println();
            
        
            cnt = 0;
        } //command1
        
        return answer;
    }
}

File: algorithm/programmers/level_1/K번째수.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    int from,to = 0;
    int target = 0;
    
    for(int i=0; i< commands.size(); i++){
        vector<int>temp ;
        from = commands[i][0]-1;
        to = commands[i][1];
        target = commands[i][2]-1;
        // from , to , target 확인
        
        for(int k = from; k<to; k++)
            temp.push_back(array[k]);
        // 반복문을 이용하여 from ~ to 벡터 복사
        
        sort(temp.begin(), temp.end());
        // 정렬
        
        answer.push_back(temp[target]);
        // 원하는 인덱스 값 정답 벡터에 push
    }
    return answer;
}